src/smtencoding/FStar.SMTEncoding.Encode.fs

```diff
@@ -978,7 +1028,7 @@ and encode_sigelt' (env:env_t) (se:sigelt) : (decls_t * env_t) =
                  (se.sigattrs |> BU.for_some is_uninterpreted_by_smt)
                  env fv t quals in
              let tname = lid.str in
-             let tsym = mkFreeV <| mk_fv (tname, Term_sort) in
+             let tsym = Option.get (try_lookup_free_var env lid) in
```

Comment:
This seems independent of the changes for reals, what's up with this one?

Reply:
It showed up as a bug of the thunking of top-level constants reveal by FStar.Real

Assuming a new nullary primitive type outside of prims was not exercised before and this case was not properly thunked/forced